elftools/elf/segments.py

```diff
@@ -119,11 +120,20 @@ def iter_notes(self):
             self.stream.seek(offset)
             # n_namesz is 4-byte aligned.
             disk_namesz = roundup(note['n_namesz'], 2)
+            # skip parsing truncated notes.
+            if offset+disk_namesz+roundup(note['n_descsz'], 2) > end:
+                break;
             note['n_name'] = bytes2str(
                 CString('').parse(self.stream.read(disk_namesz)))
             offset += disk_namesz
 
             desc_data = bytes2str(self.stream.read(note['n_descsz']))
+            if note['n_name'] == 'GNU':                                       
```

Comment:
Well I'm not tied to the linear walk, if there is a better was of doing it in python.
I guess if the enum(dict) was inverted i.e. key, values then it wouldn't be necessary.


But the real problem is that I didn't see anyway way to have multiple enum lists for this value depending on the note name.